packages/qgrid-ngx-plugins/src/lib/data-provider/request-count-once.ts

```diff
@@ -0,0 +1,29 @@
+import { Observable, of } from 'rxjs';
+import { switchMap, tap } from 'rxjs/operators';
+import { DataProviderPageServer } from './data-provider-page-server';
+import { DataProviderProcessContext } from './data-provider-process-context';
+import { DataProviderStrategy } from './data-provider-strategy';
+
+export class RequestTotalCountOnceStategy<T> implements DataProviderStrategy<T> {
+	private totalCount: number = 0;
+
+	constructor(
+		private server: Pick<DataProviderPageServer<T>, 'getTotal'>,
+	) { }
+
+	process(memo: T[], { model }: DataProviderProcessContext): Observable<T[]> {
+		if (this.totalCount > 0) {
```

Comment:
what will happen if count will be 0 from the server?

Reply:
> what will happen if count will be 0 from the server?

now it will try to request that data on every data-provider requestRows event.

Reply:
if there was empty array from server wi will try to send that request everytime. Do you think we should remove it?